packages/blade/src/components/Switch/AnimatedThumb.native.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import styled from 'styled-components/native';
+import React from 'react';
+import isNumber from 'lodash/isNumber';
+import Animated, {
+  interpolate,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+
+import { switchColors, switchMotion, switchSizes } from './switchTokens';
+import type { AnimatedThumbProps } from './types';
+import { getIn, makeBorderSize, useBreakpoint } from '~utils';
+import { useTheme } from '~components/BladeProvider';
+
+const StyledAnimatedThumb = styled(Animated.View)<{ isDisabled?: boolean }>(
+  ({ theme, isDisabled }) => {
+    const variant = isDisabled ? 'disabled' : 'default';
+    const backgroundColor = getIn(theme, switchColors.thumb[variant].background);
+
+    return {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      flexShrink: 0,
+      width: '100%',
+      height: '100%',
+      borderRadius: makeBorderSize(theme.border.radius.max),
+      backgroundColor,
+      position: 'absolute',
+    };
+  },
+);
+
+const AnimatedThumb = ({
+  isChecked,
+  isDisabled,
+  size = 'medium',
+  children,
+  isPressed,
+}: AnimatedThumbProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
```

Comment:
```suggestion
  const { theme, platform } = useTheme();
```

Reply:
Can't because the values are `'onDesktop' | 'onMobile'` while i want `'desktop' | 'mobile'` to cleanly pick the correct value from token object.